### PR TITLE
RSDK 3591 Update Workflows FIX

### PR DESCRIPTION
**Pull request by jeremyrhyde**
Fixed original PR (see link below) by adding `release_type = "pr"` to `pullrequest-trusted.yml` and casing based on it in `appimage.yml`

Previous PR: https://github.com/viamrobotics/viam-cartographer/pull/176/files

The previous failed run: https://github.com/viamrobotics/viam-cartographer/actions/runs/5446824378/workflow

### CODE REVIEW

File: .github/workflows/appimage.yml [hunk 1]
```diff
@@ -26,14 +34,14 @@ jobs:
     timeout-minutes: 30
 
     steps:
-    - name: Check out main code
-      if: github.event_name == 'workflow_dispatch' || github.event_name == 'push'
+    - name: Check out code
+      if: inputs.release_type != 'pr'
```
Comment (jeremyrhyde) [inline .github/workflows/appimage.yml:38]: [note] casing on 'pr'

File: .github/workflows/appimage.yml [hunk 2]
```diff
@@ -26,14 +34,14 @@ jobs:
...
       uses: actions/checkout@v3
       with:
         submodules: recursive
       
     - name: Check out PR branch code
-      if: github.event_name == 'pull_request_target'
+      if: inputs.release_type == 'pr'
```
Comment (jeremyrhyde) [inline .github/workflows/appimage.yml:44]: [note] casing on 'pr'

File: .github/workflows/pullrequest-trusted.yml [hunk 3]
```diff
@@ -28,14 +28,18 @@ jobs:
...
     secrets:
       GCP_CREDENTIALS: ${{ secrets.GCP_CREDENTIALS }}
 
   appimage-ignore-tests:
     if: |
        always() && !cancelled() && contains(github.event.pull_request.labels.*.name, 'safe to test') &&
        contains(github.event.pull_request.labels.*.name, 'appimage-ignore-tests')
-    uses: viamrobotics/viam-cartographer/.github/workflows/appimage.yml@main
+    uses: viamrobotics/viam-cartographer/.github/workflows/appimage.yml@RSDK-3591_UpdateWorkflows
+    with:
```
Comment (jeremyrhyde) [inline .github/workflows/pullrequest-trusted.yml:42]: [note] add 'pr' release type

**Review by jeremyrhyde — COMMENTED** [review]
Noting changes from original PR

File: .github/workflows/appimage.yml (hunk 1 above)
Comment (nicksanford) [reply .github/workflows/appimage.yml:38]: why do we need this if check if we do the same thing whether or not the release type is `pr`?

File: .github/workflows/appimage.yml [hunk 4]
```diff
@@ -52,42 +60,34 @@ jobs:
...
-        destination: 'packages.viam.com/apps/slam-servers/'
-        glob: '*'
-        parent: false
-        gzip: false
 
-    - name: Build AppImage (Latest)
-      if: github.event_name == 'workflow_dispatch' || github.event_name == 'push'
-      run: sudo -u testbot bash -lc 'make appimage-ci'
+    - name: Build and Package (PR)
+      if: inputs.release_type == 'pr'
```
[the remark this answers is not in the document]
Comment (nicksanford) [reply .github/workflows/appimage.yml:69]: what is setting this release type?

File: .github/workflows/appimage.yml (hunk 1 above)
Comment (jeremyrhyde) [reply .github/workflows/appimage.yml:38]: The difference is that the PR release_type will grab from head of the pull_request: `ref: `${{ github.event.pull_request.head.sha }}`

File: .github/workflows/appimage.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (jeremyrhyde) [reply .github/workflows/appimage.yml:69]: The other workflows, you can see in `stable.yml` we run appimage with the release_tag = stable, in `releasecandidate.yml` release_tag = `rc`, in `main.yml` release_tag = `latest` and now in in `pullrequest-trusted.yml` release_tag = `pr`

**Review by nicksanford — APPROVED** [review]

**Review by Otterverse — APPROVED** [review]
LGTM

**Pull request merged**